portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -36,14 +38,25 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the request
+    String maxCommentsString = request.getParameter("max-comments");
+
+    // Convert the input to an int or Integer.MAX_VALUE
+    int maxComments;
+    try {
+      maxComments = Integer.parseInt(maxCommentsString);
+    } catch (NumberFormatException e) {
+      maxComments = Integer.MAX_VALUE;
```

Comment:
Imagine that you've got 4 billion comments in your database and the page is requested without the max-comments parameter. I don't think it'll be easy to load all of them, and depending on their size the builder may OOM.
This is also a valid concern if max-comments is provided but it is too large.
The other concern is 'max-comments=<negative number>'.
I would suggest to have safeguards and a reasonable default number.

Reply:
Thanks for the observation! I added a limit and a default value of 10000 comments, I think this is reasonable.